applications/nrf_desktop/src/modules/config.c

```diff
@@ -234,13 +260,26 @@ static bool module_event_handler(const struct module_state_event *event)
 			req_state |= flag;
 
 			if (req_state == BIT_MASK(ARRAY_SIZE(req_modules))) {
-				init();
+				init_settings();
 			}
 
 			break;
 		}
 	}
+}
 
+static bool module_event_handler(const struct module_state_event *event)
+{
+	static bool initialized;
+
+	if (!initialized) {
+		if (event->module_id != MODULE_ID(MODULE)) {
+			load_module(event);
+		} else if (check_state(event, MODULE_ID(MODULE),
+			   MODULE_STATE_READY)) {
+			initialized = true;
```

Comment:
This is fine but we should set it when last required module is loaded.
The problem that can possibly happen is when some module will send READY twice before we get ours READY event.

Reply:
It should not occur, config module sets module state to ready if configuration is read and that may happen only after all required modules are in state ready before. Take a look into load_settings function. There is a loop that takes care about which module is already initialized.

Reply:
ok - I missed it. we do it at line 242

Reply:
in such case we don't need this case